crates/protobuf_extractor/src/extract_protobuf_imports.rs

```diff
@@ -3,44 +3,80 @@ use tree_sitter::Parser;
 
 #[derive(Debug, PartialEq)]
 pub struct ProtobufSource {
-    pub imports: Vec<String>
+    pub imports: Vec<String>,
+    pub bzl_gen_build_commands: Vec<String>,
 }
 
 impl ProtobufSource {
     pub fn parse(source: &str, _source_path: &str) -> Result<ProtobufSource> {
         let mut buf = Vec::default();
         let mut parser = Parser::new();
+        let mut bzl_gen_build_commands = Vec::default();
         parser.set_language(tree_sitter_proto::language())?;
         let tree = match parser.parse(source, None) {
             Some(tree) => tree,
-            None       => bail!("parse failed"),
+            None => bail!("parse failed"),
         };
         let root_node = tree.root_node();
         let mut cursor = root_node.walk();
         let bytes = source.as_bytes();
         for child_node in root_node.children(&mut cursor) {
             match child_node.kind() {
-                "import" =>
-                    {
-                        for path in child_node.children_by_field_name("path", &mut child_node.walk()) {
-                            let mut quoted = path.utf8_text(bytes)?.chars();
-                            // Remove the quotation marks
-                            quoted.next();
-                            quoted.next_back();
+                "import" => {
+                    for path in child_node.children_by_field_name("path", &mut child_node.walk()) {
+                        let mut quoted = path.utf8_text(bytes)?.chars();
+                        // Remove the quotation marks
+                        quoted.next();
+                        quoted.next_back();
+                        if !(quoted.as_str().starts_with("google/")) {
                             buf.push(quoted.as_str().to_string());
                         }
                     }
-                _        => ()
+                }
+                "comment" => {
+                    let raw_comment = child_node.utf8_text(bytes)?.to_string();
+                    if let Some(comment_line) = raw_comment.trim_start().strip_prefix("//") {
```

Comment:
I am not in love with the nesting. What about something like:
```rust
let opt_cmd = raw_comment
  .trim_start()
  .strip_prefix("//")
  .and_then(|comment_line| comment_line.trim_start().strip_prefix("bzl_gen_build"))
  .and_then(|matching| matching.trim_start().strip_prefix(':'));
 
if let Some(bzl_command) = opt_cmd {
  bzl_gen_build_commands.push(bzl_command.trim().to_string());
}
```

Reply:
Addressed in https://github.com/bazeltools/bzl-gen-build/pull/117/commits/e034038a27cfab319216bf41118a33c2e2fb8dce

Reply:
I've split directive related things into https://github.com/bazeltools/bzl-gen-build/pull/118